python/example_code/bedrock-agent/prompts/prompt.py

```diff
@@ -0,0 +1,234 @@
+
+    except Exception as e:
+        logger.exception("Unexpected error creating prompt version: %s", str(e))
+        raise
+# snippet-end:[python.example_code.bedrock.create_prompt_version]
+
+# snippet-start:[python.example_code.bedrock.get_prompt]
+def get_prompt(client, prompt_id):
+    """
+    Gets an Amazon Bedrock managed prompt.
+
+    Args:
+    client: Amazon Bedrock Agent boto3 client.
+    prompt_id (str): The identifier of the prompt that you want to get.
+
+    Returns:
+        dict: The response from the GetPrompt operation.
+    """
+    try:
+        logger.info("Getting prompt ID: %s.", prompt_id)
+
+        response = client.get_prompt(
+            promptIdentifier=prompt_id
+        )
+
+        logger.info("Retrieved prompt ID: %s. Name: %s", 
+                    prompt_id,
+                    response['name'])
+
+        return response
+
+    except ClientError as e:
+        logger.exception("Client error getting prompt: %s", str(e))
+        raise
+
+    except Exception as e:
+        logger.exception("Unexpected error getting prompt: %s", str(e))
+        raise
+# snippet-end:[python.example_code.bedrock.get_prompt]
+
+
+# snippet-start:[python.example_code.bedrock.delete_prompt]
+def delete_prompt(client, prompt_id, skip_resource_in_use_check=False):
+    """
+    Deletes an Amazon Bedrock managed prompt.
+
+    Args:
+    client: Amazon Bedrock Agent boto3 client.
+    prompt_id (str): The identifier of the prompt that you want to delete.
+    skip_resource_in_use_check (bool, optional): Whether to skip checking if the prompt is in use.
+                                               This parameter is ignored as it's not supported by the API.
+
+    Returns:
+        dict: The response from the DeletePrompt operation.
+    """
+    try:
+        logger.info("Deleting prompt ID: %s.", prompt_id)
+
+        response = client.delete_prompt(
+            promptIdentifier=prompt_id
+        )
+
+        logger.info("Finished deleting prompt ID: %s", prompt_id)
+
+        return response
+
+    except ClientError as e:
+        logger.exception("Client error deleting prompt: %s", str(e))
+        raise
+
+    except Exception as e:
+        logger.exception("Unexpected error deleting prompt: %s", str(e))
+        raise
+# snippet-end:[python.example_code.bedrock.delete_prompt]
+
+# snippet-start:[python.example_code.bedrock.list_prompts]
+def list_prompts(client, max_results=10):
+    """
+    Lists Amazon Bedrock managed prompts.
+
+    Args:
+        client: Amazon Bedrock Agent boto3 client.
+        max_results (int): Maximum number of results to return per page.
+
+    Returns:
+        list: A list of prompt summaries.
+    """
+    try:
+        finished = False
+        all_prompts = []
+        next_token = None
+
+        logger.info("Listing prompts:")
+
+        while not finished:
+            if next_token:
+                response = client.list_prompts(maxResults=max_results, nextToken=next_token)
```

Comment:
Is there a paginator to use here?

Reply:
Good point, I've updated to use a paginator.